android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java

```diff
@@ -16,42 +16,163 @@
-        }
-        return 0;
+
+    private DisplayManager mDisplayManager;
+    private DisplayManager.DisplayListener mListener;
+    private Display mDisplay;
+
+    /**
+     * Listens for changes in a given display device
+     */
+    public interface DisplayListener {
+        /**
+         * Called whenever the properties of a logical {@link Display}, such as size, density or
+         * refresh rate have changed.
+         */
+        void onDisplayChanged();
     }
 
-    public static long getPresentationDeadlineNanos(Display display) {
-        if (Build.VERSION.SDK_INT >= 29) {
-            return display.getPresentationDeadlineNanos();
-        }
-        return 0;
+    /**
+     * Creates a DisplayHelper which helps managing a {@link Display}.
+     *
+     * The {@link Display} to manage is specified with {@link #setListener}
+     *
+     * @param context A context to used to retrieve teh {@link DisplayManager}
+     */
+    public DisplayHelper(@NonNull Context context) {
+        mDisplayManager = (DisplayManager) context.getSystemService(Context.DISPLAY_SERVICE);
     }
 
-    public static long getRefreshPeriodNanos(Display display) {
-        return (long) (1000000000.0 / display.getRefreshRate());
+    /**
+     * Sets a callback to by notified when a given {@link Display} properties change.
+     *
+     * The callback is always automatically called once when this function is called. There can
+     * only be a single {@link DisplayListener} set.
+     *
+     * @param listener  the callback to invoke
+     * @param display   the {@link Display} to monitor changes for
+     * @param handler   an optional {@link Handler} to invoke the callback on
+     */
+    public void setListener(@NonNull DisplayListener listener, @NonNull Display display, Handler handler) {
+        mDisplay = display;
+        mListener = new DisplayManager.DisplayListener() {
+            @Override
+            public void onDisplayAdded(int displayId) {
+            }
+            @Override
+            public void onDisplayRemoved(int displayId) {
+            }
+            @Override
+            public void onDisplayChanged(int displayId) {
+                if (displayId == display.getDisplayId()) {
+                    listener.onDisplayChanged();
+                }
+            }
+        };
+        mDisplayManager.registerDisplayListener(mListener, handler);
+
+        // always invoke the callback when it's registered
+        if (handler != null) {
+            handler.post(new Runnable() {
+                @Override
+                public void run() {
+                    listener.onDisplayChanged();
+                }
+            });
+        } else {
+            listener.onDisplayChanged();
+        }
     }
 
-    public static float getRefreshRate(Display display) {
-        return display.getRefreshRate();
+    /**
+     * Unsets the {@link DisplayListener} if any
+     */
+    public void unsetListener() {
```

Comment:
On Android we use either addListener/removeListener() or setListener() but we accept null to unset it.